vega_sim/scenario/common/agents.py

```diff
@@ -1129,3 +1138,235 @@ def initialise(self, vega: VegaServiceNull):
+        self.base_order_size = base_order_size
+        self.tag = tag
+        self.momentum_strategy = momentum_strategy
+        self.indicator_threshold = indicator_threshold
+        self.trading_proportion = trading_proportion
+        self.random_state = (
+            random_state if random_state is not None else np.random.RandomState()
+        )
+        self.send_limit_order = send_limit_order
+        self.offset_levels = offset_levels
+
+        self.prices = []
+
+        self.momentum_func_dict = {
+            "RSI": self._RSI,
+            "CMO": self._CMO,
+            "STOCHRSI": self._STOCHRSI,
+            "APO": self._APO,
+            "MACD": self._MACD,
+        }
+
+        self.indicators = []
+
+    def initialise(self, vega: VegaServiceNull):
+        super().initialise(vega=vega)
+
+        self.market_id = [
+            m.id
+            for m in self.vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+
+        self.asset_id = self.vega.find_asset_id(symbol=self.asset_name)
+        self.vega.mint(
+            self.wallet_name,
+            asset=self.asset_id,
+            amount=self.initial_asset_mint,
+        )
+
+        self.pdp = self.vega._market_pos_decimals.get(self.market_id, {})
+        self.mdp = self.vega._market_price_decimals.get(self.market_id, {})
+        self.adp = self.vega._asset_decimals.get(self.asset_id, {})
+        self.vega.wait_for_total_catchup()
+
+    def step(self, vega_state: VegaState):
+        self._collect_prices()
+        signal = self.momentum_func_dict.get(self.momentum_strategy[0])()
+        if signal == 0:
+            return
+
+        trade_side = vega_protos.SIDE_BUY if signal == 1 else vega_protos.SIDE_SELL
+        volume = (
+            self.random_state.poisson(self.buy_intensity)
+            if signal == 1
+            else self.random_state.poisson(self.sell_intensity)
+        )
+        volume *= self.trading_proportion * self.base_order_size
+
+        if volume:
+            if not self.send_limit_order:
+                self.vega.submit_market_order(
+                    trading_wallet=self.wallet_name,
+                    market_id=self.market_id,
+                    side=trade_side,
+                    volume=volume,
+                    wait=False,
+                    fill_or_kill=False,
+                )
+            else:
+                best_bid, best_ask = self.vega.best_prices(market_id=self.market_id)
+                price = (
+                    best_ask + self.offset_levels / 10**self.mdp
+                    if signal == 1
+                    else best_bid - self.offset_levels / 10**self.mdp
+                )
+                self.vega.submit_order(
+                    trading_wallet=self.wallet_name,
+                    market_id=self.market_id,
+                    order_type="TYPE_LIMIT",
+                    time_in_force="TIME_IN_FORCE_IOC",
```

Comment:
Ah so we're using limit orders as market orders with fixed worst prices rather than staying on the book?

Reply:
Have changed the momentum agent to allow it multi types of limit orders, like GTC/GTT, and set the default one to be IOC.